s3/src/main/scala/akka/stream/alpakka/s3/settings.scala

```diff
@@ -90,7 +183,8 @@ final class S3Settings private (
     val s3RegionProvider: AwsRegionProvider,
     val pathStyleAccess: Boolean,
     val endpointUrl: Option[String],
-    val listBucketApiVersion: ApiVersion
+    val listBucketApiVersion: ApiVersion,
+    val forwardProxy: Option[ForwardProxy]
```

Comment:
Do you think we could reuse `proxy` which is already in the settings?

Reply:
As I understand it and from experience using the library, the existing proxy, is meant for the case where you don't want to integrate with AWS S3 but a service that implements the S3 API.  You could still have a need of using a forward proxy together with this service. E.g. company policy might dictate the need for using a forward proxy, while connecting to some third-party S3 compatible service. Leaving the old proxy settings alone means backwards compatibility, but the name is probably not ideal, since 'proxy' without any qualifications is usually a forward proxy.. Though introducing a new 'forwardProxy' should help in disambiguating :)

Reply:
I see. Yea, the current `proxy` setting controls the request `Uri`, very similar to the `endpoint-url` setting as well. There is some overlap between these two and it could be cleaned, but that's for another time.

`ForwardProxy` sounds good to me then.